contentcuration/contentcuration/utils/assessment/qti/imsmanifest.py

```diff
@@ -100,3 +104,74 @@ class Manifest(XMLElement):
     organizations: Organizations = Field(default_factory=Organizations)
     resources: Resources = Field(default_factory=Resources)
     manifests: List["Manifest"] = Field(default_factory=list)
+
+
+def _get_item_ids_from_assessment_test(zip_file, test_href):
+    """Extract assessment item identifiers from an assessment test file."""
+    try:
+        with zip_file.open(test_href) as test_file:
+            test_content = test_file.read()
+            test_root = ET.fromstring(test_content)
+
+            # Look for both item references and inline items
+            qti_ns = {"qti": "http://www.imsglobal.org/xsd/imsqti_v3p0"}
+            item_refs = test_root.findall(".//qti:qti-assessment-item-ref", qti_ns)
+            inline_items = test_root.findall(".//qti:qti-assessment-item", qti_ns)
+
+            all_items = list(item_refs) + list(inline_items)
+
+            return [
+                item.get("identifier") for item in all_items if item.get("identifier")
+            ]
+    except (KeyError, ET.ParseError):
+        return []
+
+
+namespace_re = re.compile("\\{([^}]+)\\}")
+
+
+def get_assessment_ids_from_manifest(zip_file_handle):
+    try:
+        with zipfile.ZipFile(zip_file_handle, "r") as zip_file:
+
+            # Read and parse the manifest
+            with zip_file.open("imsmanifest.xml") as manifest_file:
+                manifest_content = manifest_file.read()
+
+            # Parse the XML
+            root = ET.fromstring(manifest_content)
+
+            namespace = namespace_re.search(root.tag).group(1)
+
+            # Define namespace map for IMS Content Packaging
+            namespaces = {"imscp": namespace}
+
+            # Find all resources
+            resources = root.findall(".//imscp:resource", namespaces)
+
+            assessment_ids = []
+
+            # First, collect direct assessment item resources
+            for resource in resources:
+                resource_type = resource.get("type", "")
+                resource_identifier = resource.get("identifier")
+                if (
+                    resource_type == ResourceType.ASSESSMENT_ITEM.value
+                    and resource_identifier
+                ):
+                    assessment_ids.append(resource_identifier)
+
+                if resource_type == ResourceType.ASSESSMENT_TEST.value:
```

Comment:
I think the code here may be insufficient, as it does not handle AssessmentSections, and AssessmentSectionRefs, but I'd rather make follow up issues for these things.